backend/src/main/java/com/hackner/musicompass/controller/ArtistSearchController.java

```diff
@@ -0,0 +1,24 @@
+package com.hackner.musicompass.controller;
+
+import com.hackner.musicompass.model.Artist;
+import com.hackner.musicompass.service.ArtistSearchService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+@RequestMapping("api/artistsearch")
+public class ArtistSearchController {
+
+    private final ArtistSearchService artistSearchService;
+
+    @Autowired
+    public ArtistSearchController(ArtistSearchService artistSearchService) {
+        this.artistSearchService = artistSearchService;
+    }
+
+    @GetMapping("{searchTerm}")
```

Comment:
i would change the endpoint to "api/artist" and have query parameter search or q so an url would look like
"api/artist?q=ACDC"

Reply:
done. thx